src/LetsTrace/Samplers/RemoteControlledSampler.cs

```diff
@@ -192,6 +192,11 @@ public override bool Equals(object obj)
 
             return false;
         }
+        public override int GetHashCode()
+        {
+            // TODO Do we need special handling here? (see Equals handling)
+            return base.GetHashCode();
```

Comment:
You can use the following code. When I started the method, I haven't had the readonly flags set and GetHashCode should only contain non-writeable data.

```csharp
unchecked
{
	var hashCode = (_samplingManager != null ? _samplingManager.GetHashCode() : 0);
	hashCode = (hashCode * 397) ^ (_serviceName != null ? _serviceName.GetHashCode() : 0);
	hashCode = (hashCode * 397) ^ (_loggerFactory != null ? _loggerFactory.GetHashCode() : 0);
	hashCode = (hashCode * 397) ^ (_logger != null ? _logger.GetHashCode() : 0);
	hashCode = (hashCode * 397) ^ (_metrics != null ? _metrics.GetHashCode() : 0);
	hashCode = (hashCode * 397) ^ (_samplerFactory != null ? _samplerFactory.GetHashCode() : 0);
	hashCode = (hashCode * 397) ^ (_cancellationTokenSource != null ? _cancellationTokenSource.GetHashCode() : 0);
	hashCode = (hashCode * 397) ^ (_pollTimer != null ? _pollTimer.GetHashCode() : 0);
	return hashCode;
}
```